packages/kokkos/core/unit_test/TestDefaultDeviceTypeInit.hpp

```diff
@@ -273,7 +281,7 @@ void check_correct_initialization(const Kokkos::InitArguments& argstruct) {
 
     int expected_device = argstruct.device_id;
     if (argstruct.device_id < 0) {
-      expected_device = 0;
+      expected_device = get_ctest_gpu();
```

Comment:
@KyleFromKitware, this results in a compile error:

```
FAILED: packages/kokkos/core/unit_test/CMakeFiles/KokkosCore_UnitTest_DefaultInit_16.dir/default/TestDefaultDeviceTypeInit_16.cpp.o 
/home/projects/ppc64le-pwr9-nvidia/openmpi/2.1.2/gcc/7.2.0/cuda/9.2.88/bin/mpicxx  -DGTEST_HAS_PTHREAD=0 -I. -I/ascldap/users/rabartl/Trilinos.base/Trilinos/packages/kokkos/tpls/gtest -Ipackages/kokkos/core/src -I/ascldap/users/rabartl/Trilinos.base/Trilinos/packages/kokkos/core/src -Ipackages/kokkos -I/ascldap/users/projects/ppc64le-pwr9-nvidia/cuda/9.2.88/include -Ipackages/kokkos/core/unit_test -I/ascldap/users/rabartl/Trilinos.base/Trilinos/packages/kokkos/core/unit_test -expt-extended-lambda -arch=sm_70  -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -O3 -DNDEBUG -fPIE   -std=c++11 -MD -MT packages/kokkos/core/unit_test/CMakeFiles/KokkosCore_UnitTest_DefaultInit_16.dir/default/TestDefaultDeviceTypeInit_16.cpp.o -MF packages/kokkos/core/unit_test/CMakeFiles/KokkosCore_UnitTest_DefaultInit_16.dir/default/TestDefaultDeviceTypeInit_16.cpp.o.d -o packages/kokkos/core/unit_test/CMakeFiles/KokkosCore_UnitTest_DefaultInit_16.dir/default/TestDefaultDeviceTypeInit_16.cpp.o -c /ascldap/users/rabartl/Trilinos.base/Trilinos/packages/kokkos/core/unit_test/default/TestDefaultDeviceTypeInit_16.cpp
/ascldap/users/rabartl/Trilinos.base/Trilinos/packages/kokkos/core/unit_test/TestDefaultDeviceTypeInit.hpp(284): error: identifier "get_ctest_gpu" is undefined

1 error detected in the compilation of "/tmp/tmpxft_000041d5_00000000-6_TestDefaultDeviceTypeInit_16.cpp1.ii".
```

Any chance you can log onto 'waterman' and get this working?

For now I will commit a revert commit so that I can continue testing.

Reply:
Looks like when I tested this locally, I didn't use a CUDA compiler, which resulted in not actually compiling this bit of code, which is why I didn't catch the error. I've just added the namespace. Please try it again.

Reply:
I have this fixed now.  I just pushed https://github.com/trilinos/Trilinos/pull/6840/commits/a3c3bb83c61653f7204fdd9c369d68c34a3602b8.

I will create a Kokkos PR for this.